tests/test_drivers.py

```diff
@@ -49,5 +52,41 @@ def testMissingSubmitPath(self):
             _init_submission_driver({"payload": {"outputRun": "bad"}})
 
 
+class TestPingDriver(unittest.TestCase):
+    def testWmsServiceSuccess(self):
+        retval = ping_driver("wms_test_utils.WmsServiceSuccess")
+        self.assertEqual(retval, 0)
+
+    def testWmsServiceFailure(self):
+        with self.assertLogs(level=logging.INFO) as cm:
+            retval = ping_driver("wms_test_utils.WmsServiceFailure")
+            self.assertRegex(cm.output[0], "WARN.+service failure")
+            self.assertRegex(cm.output[1], "ERROR.+Couldn't contact service X")
```

Comment:
This is fine but another option is to look at `cm.records[1]` which is the `LogRecord` object and so can be queried separately for the log level and log message.